src/components/PopupWithForm.js

```diff
@@ -38,4 +37,9 @@ export default class PopupWithForm extends Popup {
     this._popupForm.reset();
     super.close();
   }
+
+  setLoading(buttonValue) {
```

Comment:
Better to make so, that this method will also accept `isLoading` flag, so you will be able to return the initial button text if not loading `this._submitButton.textContent = isLoading ? buttonValue : this._submitButtonValue;`